httpserver/bin/hello_world_server.dart

```diff
@@ -7,13 +7,18 @@
 
 import 'dart:io';
 
-main() {
-  HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4040)
-      .then((HttpServer server) {
-    print('listening on localhost, port ${server.port}');
-    server.listen((HttpRequest request) {
-      request.response.write('Hello, world!');
-      request.response.close();
-    });
-  }).catchError((e) => print(e.toString()));
+main() async {
+  var requestServer =
+      await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4040);
+  print('listening on localhost, port ${requestServer.port}');
+
+  try {
```

Comment:
I think you can lose the try/catch here too. It doesn't buy anything in the `async` version.